metadata/src/main/java/org/apache/kafka/controller/metrics/ControllerMetadataMetrics.java

```diff
@@ -117,6 +120,15 @@ public Integer value() {
                 return (int) zkMigrationState();
             }
         }));
+
+        if (zkMigrationEnabled) {
```

Comment:
We actually want this one reported in all cases. For de-novo KRaft clusters, this will be 0 (never migrated) which will let us distinguish from clusters which have a completed a migration (3 for post-migration). We can remove this metric in 4.0 once ZK and all the migration code is removed

Reply:
This is the wrong place to put this -- it belongs in `ControllerMetadataMetrics` since it can be calculated directly from the `MetadataImage`.

We should also do the same for `ZkMigrationState`, it should not be in `ControllerMetadataMetrics` since it comes from the image.

Separately from that, I wonder why we need to have this "if" statement. Can't we just unconditionally report this (as 0 when there no migrating brokers).